### PR TITLE
Add org.freedesktop.Sdk.Extension.freepascal

### PR DESCRIPTION
**Pull request by catsout**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [ ] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

**Comment by catsout** [comment]
bot, build org.freedesktop.Sdk.Extension.fpc

**Comment by catsout** [comment]
bot, build org.freedesktop.Sdk.Extension.fpc

**Comment by catsout** [comment]
bot, build org.freedesktop.Sdk.Extension.fpc

**Comment by catsout** [comment]
bot, build org.freedesktop.Sdk.Extension.fpc

**Comment by gasinvein** [comment]
Oh this is nice. Consider also adding lazbuild, it's probably used by most of freepascal-based apps (and probably change ID from .fpc ro .freepascal). Building lazarus can also serve as a compiler test.

**Comment by catsout** [comment]
>Oh this is nice. Consider also adding lazbuild, it's probably used by most of freepascal-based apps (and probably changing ID from .fpc to .freepascal). Building lazarus can also serve as a compiler test.

@gasinvein Sadly, maybe we can't do this. As lazarus's lcl can build against qt5 or gtk2(gtk3 in experimental), it's hard to build it on org.freedesktop.Sdk.  
Although, lazarus give a chance to rebuild lcl when needed(as source is installed), but /usr/lib/sdk/freepascal is read only...
Any idea?

**Comment by gasinvein** [comment]
I think we need only the `lazbuild` utility, not the whole IDE. If I remember, it was possible to build headless variant, with `LCL_PLATFORM=nogui` make arg or something like that.

**Comment by catsout** [comment]
It's seems ok.  
```sh
Hint: (lazarus) Missing state file of LCL 2.0.10: /usr/lib/sdk/freepascal/share/lazarus/lcl/units/x86_64-linux/gtk2/LCL.compiled
Error: (lazarus) unable to create package output directory "/usr/lib/sdk/freepascal/share/lazarus/lcl/units/x86_64-linux/gtk2" of package "LCL 2.0.10"
Hint: (lazarus) normal output directory of package LCL 2.0.10 is not writable: "/usr/lib/sdk/freepascal/share/lazarus/lcl/units/x86_64-linux/gtk2/"
TParsedCompilerOptions.SetOutputDirectoryOverride New=/home/out/.lazarus/lib/units/x86_64-linux/gtk2
Hint: (lazarus) Missing state file of LCL 2.0.10: /home/catsout/.lazarus/lib/units/x86_64-linux/gtk2/LCL.compiled
```
The `/usr/lib/sdk/freepascal/share/lazarus/lcl/units/x86_64-linux/gtk2` is read only.
But lazbuild will fallback on `$HOME/.lazarus`, and build units there.

**Comment by catsout** [comment]
bot, build org.freedesktop.Sdk.Extension.freepascal

File: org.freedesktop.Sdk.Extension.freepascal.yml [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+      - make -C fpcsrc rtl_clean rtl_smart PP=`pwd`/fpcsrc/compiler/${PPNAME}
+      # compile others with new pp,rtl
+      - make -C fpcsrc packages_smart utils_all PP=`pwd`/fpcsrc/compiler/${PPNAME}
+      # install 
+      - make -C fpcsrc compiler_distinstall rtl_distinstall packages_distinstall utils_distinstall
+        INSTALL_PREFIX=${PREFIX}
+        PP=`pwd`/fpcsrc/compiler/${PPNAME}
+        FPCMAKE=`pwd`/fpcsrc/utils/fpcm/bin/${FLATPAK_ARCH}-linux/fpcmake
+      # make cfg, add lib path, and link ppc
+      -  ${PREFIX}/lib/fpc/${fpcver}/samplecfg ${PREFIX}/lib/fpc/${fpcver} ${PREFIX}/etc &&
```
Comment (gasinvein) [inline org.freedesktop.Sdk.Extension.freepascal.yml:40]:
```suggestion
      -  ${PREFIX}/lib/fpc/${fpcver}/samplecfg ${PREFIX}/lib/fpc/${fpcver} ${PREFIX}/etc
```

**Comment by catsout** [comment]
bot, build org.freedesktop.Sdk.Extension.freepascal

**Pull request closed without merging**